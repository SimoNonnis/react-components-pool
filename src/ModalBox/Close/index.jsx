import PropTypes from 'prop-types';
import React from 'react';

import CloseModal from './style';

const ModalClose = ({ onClick }) => <CloseModal onClick={onClick}>&times;</CloseModal>;

ModalClose.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default ModalClose;
