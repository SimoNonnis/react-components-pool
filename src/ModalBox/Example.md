ModalBox with default props

```jsx
initialState = { isOpen: false };

<div>
  <button onClick={() => setState({ isOpen: true })}>Open ModalBox</button>
  <ModalBox
    isOpen={state.isOpen}
    onClose={() => setState({ isOpen: false })}
    contentLabel="Demo Intro"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
  </ModalBox>
</div>;
```

ModalBox without close label

```jsx
initialState = { isOpen: false };

<div>
  <button onClick={() => setState({ isOpen: true })}>Open ModalBox</button>
  <ModalBox
    isOpen={state.isOpen}
    onClose={() => setState({ isOpen: false })}
    contentLabel="Demo Intro"
    noCloseLabel
  >
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
  </ModalBox>
</div>;
```

ModalBox with a lot of content

```jsx
initialState = { isOpen: false };

<div>
  <button onClick={() => setState({ isOpen: true })}>Open ModalBox</button>
  <ModalBox
    isOpen={state.isOpen}
    onClose={() => setState({ isOpen: false })}
    contentLabel="Demo Intro"
    tall
  >
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusantium consequatur, sit
      doloremque possimus temporibus laborum voluptate aut, neque, rem hic libero quo facere magni.
      Nemo quod nulla culpa natus esse!
    </p>
  </ModalBox>
</div>;
```
