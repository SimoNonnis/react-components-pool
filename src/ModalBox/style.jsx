import styled from 'styled-components';
import ReactModal from 'react-modal';
import { media } from 'utils';

export const StyledModal = styled(ReactModal)`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  align-items: ${({ tall }) => (tall ? 'flex-start' : 'center')};
  justify-content: center;
  background-color: rgba(29, 49, 65, 0.85);
  overflow-x: hidden;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  padding: 1.2rem;
  z-index: 1031;
`;
StyledModal.displayName = 'StyledModal';

export const Container = styled.div`
  background-color: ${({ theme }) => theme.color.white};
  border-radius: 4px;
  max-width: 100%;
  padding: 28px 20px;
  ${media.phone`
    max-width: 700px;
  `};
  ${media.desktop`
    padding: 30px;
  `};
  position: relative;
`;
Container.displayName = 'Container';
