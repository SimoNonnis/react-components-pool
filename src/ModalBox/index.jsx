import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';

import ModalClose from './Close';

import { StyledModal, Container } from './style';

const propTypes = {
  isOpen: PropTypes.bool.isRequired,
  contentLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func,
  className: PropTypes.string,
  tall: PropTypes.bool,
  noCloseLabel: PropTypes.bool,
  style: PropTypes.shape({
    overlay: PropTypes.shape({}),
    content: PropTypes.shape({}),
  }),
};

const defaultProps = {
  className: '',
  onClose: noop,
  tall: false,
  noCloseLabel: false,
  style: {
    overlay: {
      zIndex: '99999',
    },
  },
};

const ModalBox = ({
  isOpen,
  contentLabel,
  children,
  onClose,
  className,
  tall,
  noCloseLabel,
  style,
  ...props
}) => (
  <StyledModal
    style={style}
    isOpen={isOpen}
    contentLabel={contentLabel}
    onRequestClose={onClose}
    className={className}
    tall={tall}
    {...props}
  >
    <Container className={className}>
      {noCloseLabel || <ModalClose onClick={onClose} />}
      {children}
    </Container>
  </StyledModal>
);

ModalBox.propTypes = propTypes;
ModalBox.defaultProps = defaultProps;

export default ModalBox;
