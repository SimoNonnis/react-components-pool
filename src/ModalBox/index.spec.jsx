/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import ModalBox from './';

describe('Test <ModalBox />', () => {
  const mockOnClick = jest.fn();
  it('should render Modal correctly with required props', () => {
    const wrapper = shallow(
      <ModalBox isOpen={false} contentLabel="Demo Intro">
        children
      </ModalBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Modal correctly with isOpen prop set to true', () => {
    const wrapper = shallow(
      <ModalBox isOpen contentLabel="Demo Intro">
        children
      </ModalBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Modal passing onClose prop', () => {
    const wrapper = shallow(
      <ModalBox isOpen={false} onClose={mockOnClick} contentLabel="Demo Intro">
        children
      </ModalBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Modal passing tall boolean prop', () => {
    const wrapper = shallow(
      <ModalBox isOpen={false} onClose={mockOnClick} tall contentLabel="Demo Intro">
        children
      </ModalBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Modal passing noX boolean prop', () => {
    const wrapper = shallow(
      <ModalBox isOpen={false} onClose={mockOnClick} noX contentLabel="Demo Intro">
        children
      </ModalBox>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
