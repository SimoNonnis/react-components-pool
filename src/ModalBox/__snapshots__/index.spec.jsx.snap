// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ModalBox /> should render Modal correctly with isOpen prop set to true 1`] = `
<StyledModal
  className=""
  contentLabel="Demo Intro"
  isOpen={true}
  onRequestClose={[Function]}
  style={
    Object {
      "overlay": Object {
        "zIndex": "99999",
      },
    }
  }
  tall={false}
>
  <Container
    className=""
  >
    <ModalClose
      onClick={[Function]}
    />
    children
  </Container>
</StyledModal>
`;

exports[`Test <ModalBox /> should render Modal correctly with required props 1`] = `
<StyledModal
  className=""
  contentLabel="Demo Intro"
  isOpen={false}
  onRequestClose={[Function]}
  style={
    Object {
      "overlay": Object {
        "zIndex": "99999",
      },
    }
  }
  tall={false}
>
  <Container
    className=""
  >
    <ModalClose
      onClick={[Function]}
    />
    children
  </Container>
</StyledModal>
`;

exports[`Test <ModalBox /> should render Modal passing noX boolean prop 1`] = `
<StyledModal
  className=""
  contentLabel="Demo Intro"
  isOpen={false}
  noX={true}
  onRequestClose={[MockFunction]}
  style={
    Object {
      "overlay": Object {
        "zIndex": "99999",
      },
    }
  }
  tall={false}
>
  <Container
    className=""
  >
    <ModalClose
      onClick={[MockFunction]}
    />
    children
  </Container>
</StyledModal>
`;

exports[`Test <ModalBox /> should render Modal passing onClose prop 1`] = `
<StyledModal
  className=""
  contentLabel="Demo Intro"
  isOpen={false}
  onRequestClose={[MockFunction]}
  style={
    Object {
      "overlay": Object {
        "zIndex": "99999",
      },
    }
  }
  tall={false}
>
  <Container
    className=""
  >
    <ModalClose
      onClick={[MockFunction]}
    />
    children
  </Container>
</StyledModal>
`;

exports[`Test <ModalBox /> should render Modal passing tall boolean prop 1`] = `
<StyledModal
  className=""
  contentLabel="Demo Intro"
  isOpen={false}
  onRequestClose={[MockFunction]}
  style={
    Object {
      "overlay": Object {
        "zIndex": "99999",
      },
    }
  }
  tall={true}
>
  <Container
    className=""
  >
    <ModalClose
      onClick={[MockFunction]}
    />
    children
  </Container>
</StyledModal>
`;
